### DISCUSSION TITLE
How can I call the method with annotation?

### DISCUSSION THREAD

**Discussion by vincentgwzhang** [junit-team/junit-framework#3421] [Q&A]
Suppose I have an annotation:

```java
@WithMockUser
@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(factory = AAFactory.class)
public @interface AppUser {
    String username() default "";
    String password() default "";
    String role() default "";
}
```

Now I annotate several test methods with `@AppUser`, like:

```java
@Test
@AppUser(username = "a1", password= "b1", role = "manager")
void testMethof1() {}

@Test
@AppUser(username = "c1", password= "d1", role = "developer")
void testMethof1() {}
```

So now can I merge them into one, like:

```java
@Test
@AppUser(username = "a1", password= "b1", role = "manager")
@AppUser(username = "c1", password= "d1", role = "developer")
void testMethof1() {}
````

or 

```java
@Test
@AppUsers ({
    @AppUser(username = "a1", password= "b1", role = "manager"),
    @AppUser(username = "c1", password= "d1", role = "developer")
})
void testMethof1() {}
```

so everything could be simplify?

**Comment by JojOatXGME** [comment]
This question misses important details. What shall happen when you add two of these annotations to one method? It is also unclear to me what you want to know exactly. As this annotation is not provided by JUnit, JUnit is also not responsible for it. Is this question about how to define annotations in Java, or is JUnit somehow relevant for this question?

PS: The code blocks in your question are also not properly formatted.

**Comment by sbrannen** [comment]
I've edited your comment to improve the formatting. You might want to check out this [Mastering Markdown guide](https://guides.github.com/features/mastering-markdown/) for future reference.

**Comment by sbrannen** [comment]
Thanks for getting in touch, but it feels like this is a question that would be better suited to [Stack Overflow](http://stackoverflow.com/).

We prefer to use the "discussions" feature of GitHub only for discussions pertaining to JUnit 5. Feel free to update this discussion with a link to the re-posted question (so that other people can find it).
